mainapp/urls.py

```diff
@@ -6,4 +6,5 @@
     path("", views.index, name="index"),
     path("accounts/", include("allauth.urls")),
     path("account", views.account, name="account"),
+    path("favlist", views.favlist, name="favlist"),
```

Comment:
while this works, I think having the location favorite list be a part of the mainapp package doesn't necessarily make a lot of sense. The favorite list isn't in reference to the mainapp, but to the locations. This is kinda nit-picky but I think having it be `/locations/favlist` or something like that makes more sense, because then it's the favorites list of locations

Reply:
For me, I would say the favorite list make sense to have it under the mainapp because a user can have many favorite locations, but it's okay to not under the mainapp but have its own application. So ideally it would be `siteurl/favlist`.

Reply:
I disagree.  Having the favorite's list under mainapp violates REST API principles of having each API endpoint map to the domain object that it is operating on.  the favorites list is a list of favorite **locations** given a **user**.  The object being operated on is the location and the variable input is the user.

Reply:
Yeah, I also couldn't make up my mind about placing it in the `mainapp` or `location` apps. But I guess it makes more sense for the favorites to be in the location app. Moved the files from `mainapp` to `location`.

Reply:
that is a good point.  if request.user.favorites is the current implementation then lets go with that.